Resources/Prototypes/_NF/XenoArch/Effects/utility_effects.yml

```diff
@@ -0,0 +1,110 @@
+# Utility effects permanently modify the entity in some way when triggered, and they generally make it 'useful' for some purpose,
+# like turning the artifact into a tool, or gun, or whatever.
+- type: artifactEffect
+  id: EffectReclaimer
+  targetDepth: 3
+  effectHint: artifact-effect-hint-multitool
+  blacklist:
+    components:
+    - Item
+  permanentComponents:
+  - type: MaterialReclaimer
+    whitelist:
+      components:
+      - PhysicalComposition
+      - SpaceGarbage
+      - Log # Frontier
+      tags:
+      - Trash
+      - Recyclable
+    blacklist:
+      components:
+      - Material
+      - Pda
+      - IdCard
+      - Brain
+      tags:
+      - HighRiskItem
+    soundCooldown: 0
+    sound:
+      path: /Audio/Ambience/Objects/crushing.ogg
+      params:
+        volume: 5
+        maxDistance: 5
+        loop: true
+    useOldSolutionLogic: true
+    solutionContainerId: output
+  - type: MaterialStorage
+    insertOnInteract: false
+  - type: SolutionContainerManager
+    solutions:
+      output:
+        maxVol: 1000 # Frontier 100<1000
+  - type: DrainableSolution
+    solution: output
+  - type: ExaminableSolution
+    solution: output
+  - type: MaterialReclaimerMagnetPickup # Frontier
+    range: 0.30
+    magnetEnabled: true
+
+- type: artifactEffect
+  id: EffectEggLayer
+  targetDepth: 2
+  effectHint: artifact-effect-hint-creation
+  blacklist:
+    components:
+    - Item
+  permanentComponents:
+  - type: EggLayer
+    eggSpawn:
+    - id: FoodEgg
+    hungerUsage: 0
+    eggLayCooldownMin: 600
+    eggLayCooldownMax: 900
```

Comment:
Why? Can we make it a temporary component instead of a permanent one, so we don't have a free infinite egg generator (or several)? You _know_ people will literally farm this shit.

Reply:
Sure ill edit